testing/src/_consistency_checker.py

```diff
@@ -476,7 +478,7 @@ def check_secrets_consistency(
         return Results(errors, [])
 
     assert event.secret is not None
-    if event.secret not in state.secrets:
+    if event.secret.id not in {s.id for s in state.secrets}:
         secret_key = event.secret.id if event.secret.id else event.secret.label
```

Comment:
if there is a chance that secret.id could be None (seems implied by this line) then we have to be careful about putting them in a set comprehension? or perhaps this line is old code that can now be removed

Reply:
I'll check on this, but I believe the event's secret should always have an ID and we seem to type it that way. Could indeed be older code.